react-icon-pack/src/Icon.tsx

```diff
@@ -1,5 +1,25 @@
-import { type IconComponent } from '../../types';
+import React from 'react';
+import * as BrokenIcons from '../dist/icons/broken';
+import * as OutlineIcons from '../dist/icons/outline';
+import * as CurvedIcons from '../dist/icons/curved';
+import { convertToPascalCase } from './utils';
+import { IconProps } from '../../types';
 
-export const Icon: IconComponent = () => {
-  return <div>Test web component</div>
+const iconVariants = {
+  broken: BrokenIcons,
+  outline: OutlineIcons,
+  curved: CurvedIcons,
+};
+
+type IconComponentName = keyof (typeof BrokenIcons | typeof OutlineIcons | typeof CurvedIcons);
```

Comment:
This type can be imported directly from project root types

Reply:
Not exactly - it is a a react icon component name type rather than icon name type from glyphmap